packages/core/src/middleware/koa-log.ts

```diff
@@ -10,17 +10,52 @@ export type WithLogContext<ContextT> = ContextT & {
   log: <T extends LogType>(type: T, payload: LogPayloads[T]) => void;
 };
 
-const saveLog = async (type: LogType, payload: LogPayload) => {
-  try {
-    await insertLog({
-      id: nanoid(),
-      type,
-      payload,
-    });
-  } catch (error: unknown) {
-    console.error('An error occurred while inserting log');
-    console.error(error);
-  }
+const initLog = () => {
+  // eslint-disable-next-line @silverhand/fp/no-let
+  let type: Optional<LogType>;
+  // eslint-disable-next-line @silverhand/fp/no-let
+  let payload: LogPayload = {};
+
+  const mergeLog = (logType: LogType, logPayload: LogPayload) => {
+    if (type !== logType) {
+      // eslint-disable-next-line @silverhand/fp/no-mutation
+      payload = {}; // Reset payload when type changes
+    }
+
+    // eslint-disable-next-line @silverhand/fp/no-mutation
+    type = logType; // Use first initialized log type
+    // eslint-disable-next-line @silverhand/fp/no-mutation
+    payload = deepmerge(payload, logPayload);
+  };
+
+  const mergeLogError = (error: unknown) => {
+    if (type) {
+      mergeLog(type, { error: String(error) });
+    }
+  };
+
+  const saveLog = async (ctx: ParameterizedContext) => {
+    if (!type) {
+      return;
+    }
+
+    const result = payload.error ? LogResult.Error : LogResult.Success;
+    const {
+      ip,
+      headers: { 'user-agent': userAgent },
+    } = ctx.request;
+
+    mergeLog(type, { result, ip, userAgent });
+
+    try {
+      await insertLog({ id: nanoid(), type, payload });
+    } catch (error: unknown) {
+      console.error('An error occurred while inserting log');
+      console.error(error);
+    }
```

Comment:
why do we need catch error here? I think we have error handler that wraps the whole app?

Reply:
just catch the error from inserting log for not blocking API request